src/ica_api.c

```diff
@@ -4177,6 +4177,32 @@ int ica_get_msa_level(void)
 	return msa;
 }
 
+int ica_get_hw_info(libica_hw_info *hw_info)
+{
+	char line[32];
+	char dummy[32];
+	FILE *fd;
+
+	if (hw_info == NULL)
+		return EINVAL;
+
+	if ((fd = fopen("/proc/cpuinfo", "r")) == NULL)
```

Comment:
Well, this function is nonsense in my opinion.  Why adding a function to libica for something that can also be done out of the box with a simple `cat /proc/cpuinfo` ?
This function does not really add anything.

Reply:
You are completely right, but fips certification requires this info to be available programmatically through the library API.

Reply:
Hmm... OpenSSL seems to be FIPS certified, right? So where in OpenSSL is this API that you can programmatically query such information?????

Reply:
The difference between openssl and libica is (as seen by the fips certifiers), that openssl is considered as a software module, while libica is considered as a "hybrid module", consisting of a software part (our libica code) and a hardware/firmware part, which is CPACF. For this reason they want the library to provide version levels of both parts. 

I already asked CPACF development whether there is an unambiguous version indication. They told me there's none. So all we have is the MSA level, which is not unique, because there are multiple processor generations with the same (highest) MSA level. But together with the processor id (machine generation) it sufficient for the certification.